Source/WebKit/WebProcess/WebProcess.cpp

```diff
@@ -1668,6 +1673,10 @@ void WebProcess::pageWillLeaveWindow(PageIdentifier pageID)
         if (!m_nonVisibleProcessMemoryCleanupTimer.isActive())
             m_nonVisibleProcessMemoryCleanupTimer.startOneShot(nonVisibleProcessMemoryCleanupDelay);
 #endif
+
+#if ENABLE(PURGEABLE_DECODED_IMAGE_DATA_CACHE)
+        PurgeableCachedDecodedImage::startCleanup();
+#endif
```

Comment:
Why do we need to add a new timer for cleaning up the images? Can't we use one of the existing timers instead: `m_nonVisibleProcessEarlyMemoryCleanupTimer` or `m_nonVisibleProcessMemoryCleanupTimer`?

Reply:
Unlike the existing timers, this one needs to be able to run again, later, if the page is still in the background but we draw images. This can happen for example when you move the pointer over the tab strip and we paint the page thumbnails. At that point we should throw away the NativeImages again shortly after. Otherwise it's easy for this patch not to have a real world effect, since the user will probably do this at some point and all the memory usage will come back.